lib/routes/routeBackbeat.js

```diff
@@ -57,9 +59,29 @@ function _getRequestPayload(req, cb) {
     .on('end', () => cb(null, Buffer.concat(payload, payloadLen).toString()));
 }
 
+function _checkErrorResponse(err, request, response, log) {
+    if (err) {
+        return responseJSONBody(err, null, response, log);
+    }
+    log.debug('backbeat route response sent successfully', {
+        method: request.method,
+        bucketName: request.bucketName,
+        objectKey: request.objectKey,
+    });
+    return undefined;
+}
+
+function _isValidAWSLocation(storageLocation) {
+    const { locationConstraints } = config;
+    return storageLocation &&
+        locationConstraints[storageLocation] &&
+        locationConstraints[storageLocation].type === 'aws_s3';
+}
+
 /*
 PUT /_/backbeat/metadata/<bucket name>/<object key>
 PUT /_/backbeat/data/<bucket name>/<object key>
+PUT /_/backbeat/aws/<bucket name>/<object key>
```

Comment:
If I understand the design correctly (correct me if I'm wrong), this route shall be agnostic of the backend since it will be driven by location constraints, so it should not be named after AWS directly but have a more generic name? Could be `/_/backbeat/object/...`, for example, meant to be used as a single-call object interface.

Reply:
Updated to `/_/backbeat/multipleBackendData/<bucket name>/<object key>`.